src/repo/mod.rs

```diff
@@ -241,20 +255,34 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
     }
 
     /// Remove block from the block store.
-    pub async fn remove_block(&self, cid: &Cid) -> Result<(), Error> {
+    pub async fn remove_block(&self, cid: &Cid) -> Result<Cid, Error> {
         let cid = cid.as_upgraded_cid();
         if self.is_pinned(&cid).await? {
             return Err(anyhow::anyhow!("block to remove is pinned"));
         }
+
         // sending only fails if the background task has exited
         self.events
             .clone()
             // unprovide only cidv1
             .send(RepoEvent::UnprovideBlock(cid.clone()))
             .await
             .ok();
-        self.block_store.remove(&cid).await?;
-        Ok(())
+
+        // FIXME: Need to change location of pinning logic.
+        // I like this pattern of the repo abstraction being some sort of
+        // "clearing house" for the underlying result enums, but this
+        // could potentially be pushed out out of here up to Ipfs, idk
+        match self.block_store.remove(&cid).await? {
+            Ok(success) => match success {
+                BlockRm::Removed(cid) => Ok(cid),
+                // TODO: Add DownloadCancelled here
```

Comment:
these todos can be removed as if you go and add a variant, this will start giving out a compiler error